perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/RegistrarManagerMethod.java

```diff
@@ -1126,25 +1130,25 @@ public Object call(ApiCaller ac, Deserializer parms) throws PerunException {
 		@Override
 		public Boolean call(ApiCaller ac, Deserializer parms) throws PerunException {
 
-			ReCaptchaImpl reCaptcha = new ReCaptchaImpl();
+			String secret = BeansUtils.getCoreConfig().getRecaptchaPrivateKey();
+			String response = parms.readString("response");
 
-			reCaptcha.setPrivateKey(BeansUtils.getCoreConfig().getRecaptchaPrivateKey());
-			reCaptcha.setRecaptchaServer(ReCaptchaImpl.HTTPS_SERVER);
+			RestTemplate restTemplate = new RestTemplate();
 
-			// we don't need caller's address since our key is global
-			String remoteAddress = "";
-			ReCaptchaResponse reCaptchaResponse = reCaptcha.checkAnswer(remoteAddress, parms.readString("challenge"), parms.readString("response"));
+			final String verificationUrl = "https://www.google.com/recaptcha/api/siteverify";
 
-			if (reCaptchaResponse.isValid()) {
-				return true;
-			} else {
-				return false;
-			}
+			HttpHeaders headers = new HttpHeaders();
+			headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
 
+			MultiValueMap<String, String> map = new LinkedMultiValueMap<>();
+			map.add("response", response);
+			map.add("secret", secret);
 
+			HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(map, headers);
 
+			JSONObject answer = new JSONObject(restTemplate.postForObject(verificationUrl, request , String.class));
```

Comment:
It can be written as 
`return restTemplate.postForObject(verificationUrl, request , JsonNode.class).path("success").asBoolean()`

Reply:
Ok, I will change it.